pkg/utils/mco_deploy.go

```diff
@@ -629,6 +629,65 @@ func CheckMCOAddon(opt TestOptions) error {
 	return nil
 }
 
+func CheckMCOAddonResources(opt TestOptions) error {
+	client := NewKubeClient(
+		opt.HubCluster.MasterURL,
+		opt.KubeConfig,
+		opt.HubCluster.KubeContext)
+	if len(opt.ManagedClusters) > 0 {
+		client = NewKubeClient(
+			opt.ManagedClusters[0].MasterURL,
+			opt.ManagedClusters[0].KubeConfig,
+			"")
+	}
+
+	deployList, err := client.AppsV1().Deployments(MCO_ADDON_NAMESPACE).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	resMap := make(map[string]corev1.ResourceRequirements)
+	for _, deploy := range deployList.Items {
+		resMap[deploy.Name] = deploy.Spec.Template.Spec.Containers[0].Resources
+	}
+
+	metricsCollectorRes := corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			"cpu":    resource.MustParse("200m"),
+			"memory": resource.MustParse("700Mi"),
+		},
+		Requests: corev1.ResourceList{
+			"cpu":    resource.MustParse("200m"),
+			"memory": resource.MustParse("200Mi"),
+		},
+	}
+
+	endpointObsOperatorRes := corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			"cpu":    resource.MustParse("100m"),
+			"memory": resource.MustParse("600Mi"),
+		},
+		Requests: corev1.ResourceList{
+			"cpu":    resource.MustParse("100m"),
+			"memory": resource.MustParse("100Mi"),
+		},
```

Comment:
Do we really need to check it? Can you imagine any cases that lead to it is not equal to default value?

Reply:
check it just to ensure we have defined resource here: https://github.com/open-cluster-management/multicluster-observability-operator/blob/main/manifests/endpoint-observability/operator.yaml#L22

Reply:
as you say. we never change this value. so I think we can remove this.